### ISSUE TITLE
`ComposeMultipleContentEmitters` not working with conditional logic.

### ISSUE THREAD

**Issue by blipinsk** [slackhq/compose-lints#372]
![CleanShot 2024-07-17 at 13 03 54](https://github.com/user-attachments/assets/1c2099c7-046c-437d-b6a3-9299ab232412)

```kotlin
// 👇 false negative
@Composable
private fun TestWithIf() {
  if (someCondition()) {
    Text("1")
    Text("2")
  } else {
    Text("2")
    Text("3")
  }
}

// 👇 correctly showing a warning
@Composable
private fun TestNoIf() {
  Text("1")
  Text("2")
}
```

**Comment by ZacSweers** [comment]
We can only do so much when it comes to complexity I think. We had a similar case with detecting if we're within a for loop. I think it would be better to separate your composables into different functions in this case
